### DISCUSSION TITLE
formulasUpdated listener

### DISCUSSION THREAD

**Discussion by BrianHung** [handsontable/hyperformula#1155] [Feature requests]
#845 #135 https://github.com/handsontable/handsontable/issues/8114

It would be useful to have a `formulasUpdated` listener, as mentioned in #135.

I've been trying to integrate hyperformula with an external data source (to work with React etc), and have noticed that some methods like `addRow` and `moveRows` modify internal formula representations. 

Essentially the same problem that handsontable faces in https://github.com/handsontable/handsontable/issues/8114, where you can move rows and columns but formulas referencing values in those moved rows or columns are now out of date.

For example if we had a `=SUM(A1:B3)` and the third row was deleted, the formula would be changed to `=SUM(A1:B2)`.

Right now the only way to correctly react to changes in formulas is to use `getAllSheetsSerialized`. 

However, this isn't as efficient as `valuesUpdated` is for listening to changes in values, since there can be a lot of sheets and would cause excessive re-rendering without manual diffs (which is also expensive).

---
For fun, you can try applying https://features.jsomers.net/how-i-reverse-engineered-google-docs/ to google sheets, and can see that they update formulas on things like moveRow too!

```js
[
    [
        24502104,
        "[\"0\",10,1,0,0]" // move 1 row to row 10
    ],
    [
        25813757,
        "[[\"0\",1,2,1,2],[null,null,[[[[[0,10,1,2,1118464],3]]]]],null,[[null,8,null,null,0]]]" // update =SUM(C2:C7) at [rowStart 1, rowEnd 2, colStart 1, colEnd 2] to be =SUM(C2:C10)
    ],
    [
        24549788,
        "[\"0\",\"0\",[[[6,7,10]]],[[[null,null,0]]]]" // move rows in [6, 7) to rowStart 10 
    ],
    [
        21299578,
        "[[\"0\",10,11,0,26],[48630543],[]]" // rowStart 10, rowEnd 11, colStart A, colEnd Z
    ],
    [
        21299578,
        "[[\"0\",6,7,0,26],[1206345535],[]]" // rowStart 6, rowEnd 7, colStart A, colEnd Z
    ],
    [
        25037233,
        "[\"0\",6,1,0]" // move 1 row from row 6
    ],
    [
        21299578,
        "[[\"0\",1,2,1,2],[null,8,null,null,0],[null,null,[[[[[0,9,1,2,1118464],3]]]]]]"
    ]
]
```

**Comment by AMBudnik** [comment]
@BrianHung Can I ask you to share the feedback at https://github.com/handsontable/hyperformula/issues/1156? I wouldn't want to loose as it is harder to track updates to GitHub Discussions.

**Reply by BrianHung** [reply]
Done!

**Reply by AMBudnik** [reply]
Thanks
